packages/paste-website/src/pages/api/share-search-usage.ts

```diff
@@ -0,0 +1,157 @@
+import { createClient } from "@supabase/supabase-js";
+import type { NextApiRequest, NextApiResponse } from "next";
+import Rollbar from "rollbar";
+import { z } from "zod";
+
+import { logger } from "../../functions-utils/logger";
+
+const queryCountSchema = z.object({
+  // eslint-disable-next-line camelcase
+  query_string: z.string(),
+  count: z.number(),
+});
+
+const latestQueriesDataSchema = z.array(queryCountSchema);
+
+const rollbar = new Rollbar({
+  accessToken: process.env.ROLLBAR_ACCESS_TOKEN,
+  captureUncaught: true,
+  captureUnhandledRejections: true,
+});
+
+const supabaseUrl = process.env.SUPABASE_URL;
+const supabaseServiceKey = process.env.SUPABASE_KEY;
+const slackChannelID = process.env.SLACK_CHANNEL_HELP_DESIGN_SYSTEM;
+
+const LOG_PREFIX = "[/api/share-search-usage]:";
+
+export default async function handler(req: NextApiRequest, res: NextApiResponse): Promise<void | Response> {
+  logger.info(`${LOG_PREFIX} Incoming request`);
+
+  // if this doesn't get called from a vercel cron job, then it's unauthorized
+  if (req.headers.authorization !== `Bearer ${process.env.CRON_SECRET}`) {
+    res.status(401).end("Unauthorized");
+    return;
+  }
+
+  if (supabaseUrl === undefined) {
+    logger.error(`${LOG_PREFIX} Supabase URL is undefined`);
+    rollbar.error(`${LOG_PREFIX} Supabase URL is undefined`);
+    res.status(500).json({
+      error: "Supabase URL is undefined",
+    });
+    return;
+  }
+
+  if (supabaseServiceKey === undefined) {
+    logger.error(`${LOG_PREFIX} Supabase service key is undefined`);
+    rollbar.error(`${LOG_PREFIX} Supabase service key is undefined`);
+    res.status(500).json({
+      error: "Supabase service key is undefined",
+    });
+    return;
+  }
+
+  if (slackChannelID === undefined) {
+    logger.error(`${LOG_PREFIX} Slack channel ID is undefined`);
+    rollbar.error(`${LOG_PREFIX} Slack channel ID is undefined`);
+    res.status(500).json({
+      error: "Slack channel ID is undefined",
+    });
+    return;
+  }
+
+  const { host } = req.headers;
+  const protocol = host?.includes("localhost") ? "http" : "https";
+
+  const supabaseClient = createClient(supabaseUrl, supabaseServiceKey);
+
+  const latestSearchQueriesResponse = await supabaseClient.rpc("match_weekly_search_usage");
+
+  if (latestSearchQueriesResponse.error) {
+    logger.error(`${LOG_PREFIX} Error fetching latest search queries`, {
```

Comment:
suggestion: I almost wish logger.error also called rollbar.error internally. Would remove a lot of this duplication code

Reply:
Yeah, it's just a instantiation of a winston logger, we might be able to create a custom transporter that calls rollbar on error maybe? No sure if that's its intended purpose though.